app/Http/Controllers/ListingController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ListPostRequest;
+use App\Models\Listing;
+use App\Models\Shop;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+
+class ListingController extends Controller
+{
+
+    public function index()
+    {
+        $listings = Listing::all();
+        return view('main.include.listing.listings', compact('listings'));
+    }
+
+
+    public function create()
+    {
+        return view('main.include.listing.listing-create');
+    }
+
+
+    public function store(ListPostRequest $request)
+    {
+        $userId = auth()->id();
+        $shop = Shop::whereUserId($userId)->first();
+        if (!is_null($request->image)) {
+            $imageName = "iSotStore_" . rand(0, 10000) . "." . $request->image->getClientOriginalExtension();
+            $request->image->move(public_path('main/images/listing'), $imageName);
+        } else {
+            $imageName = $request->image;
+        }
+        $listing = Listing::create([
+            'name' => $request->name,
+            'description' => $request->description,
+            'image'=> $imageName,
+            'unit' => $request->unit,
+            'type'=> $request->type,
+            'status' => $request->status,
+            'delivery_status' => $request->delivery_status,
+            'faulty' => $request->faulty,
+            'origin'=> $request->origin,
+            'shop_id' => $shop->id,
+            'category_id'=> $request->category_id
+        ]);
+        if($listing)
```

Comment:
1)burada listing oluşturdu mu diye if kontrolu yapmak yerine daha iyi bir yöntem var. try-catch yapısını kullanabilirsin.
try {
bütün işlemlerin
...
return "başarı ile oluşturuldu"
}
catch {
return "işlem başarısız"
}

başka yerlerde de aynısını yapmışsın onlarıda try-catch yapısına alırsın.

2)Fotolara review yapamadığım için buraya yazıyorum
public/main/images/listing/iSotStore_4255.jpg -> Bu sanırım kullanıcının girmiş olduğu bir fotoğraf. Eğer öyle ise github'a gelmemesi lazım bu dosyayı .gitignore'a alalım

Reply:
Fotoğrafları hallettim. Diğer kısıma bakacağım

Reply:
Try catch olayını bir kaç yerde kullandım, tüm projeye geçirmeye devam ettiğimden şuanlık bunu resolve olarak değiştiriyorum.